### ISSUE TITLE
Previewer is hard-coded to use PdfLaTeX

### ISSUE THREAD

**Issue by numberZero** [texstudio-org/texstudio#302]
I use XeLaTeX to compile my documents, so they contain some special code (like font selection). But internal previewer is hard-coded to use either PdfLaTeX or plain LaTeX. It works fine if I set the “PdfLaTeX” option to `xelatex`, though, but that makes me unable to use true PdfLaTeX for documents that require that.

### Environment
- TeXstudio: 2.12.10
- Qt: 5.11.1
- OS: Linux (Arch)
- TeX distribution: texlive (packages)

**Comment by septatrix** [comment]
Yes I also noticed this when using LuaLaTeX with fontspec and is something which should definitely be dynamic. Possibly the cleanest solution would be to just use the program for the current document (so we obey the default program from the settings and also make it overwriteable using the magic comments).

Is this were the preview logic is implemented? https://github.com/texstudio-org/texstudio/blob/fc2c012843150aa3185a3b8df47263ce7dc4b6e4/src/buildmanager.cpp#L1788

**Comment by e-dervieux** [comment]
I do not know whether this is linked or not, but the fact that the previewing tool is different from the compilation tool might cause some strange behaviour like this one: https://tex.stackexchange.com/questions/718530/pdftex-log-file-eating-up-disk-space

Mayhap automatically changing the preview tool so that it's identical to the compilation tool would avoid this kind of problem?

**Comment by sunderme** [comment]
more options are available

**Comment by e-dervieux** [comment]
@sunderme what exactly do you mean by that? If you mean that the user can choose between different visualization tools, this has been the case for quite some time, *cf.*

![image](https://github.com/user-attachments/assets/0e42d14a-09f2-4d1e-8f10-cbe8898b6c3b)

the point I mentioned above was more about having TeXStudio automatically switch to the pre-visualization tool corresponding to the compilation toolchain, *i.e.* choosing lualatex previz when compiling with lualatex, xelatex previz when compiling with xelatex, *etc.*

I reckon that this latter point is not exactly what the OP asked for though. Do you think I should open a new issue / enhancement for these latter considerations?

**Comment by sunderme** [comment]
The issue is quite old.
At that time only pdflatex was offered.
I think there is another feature request for automatic change already.

**Comment by e-dervieux** [comment]
> The issue is quite old. At that time only pdflatex was offered. I think there is another feature request for automatic change already.

Very much indeed, and I am the one who created it, I forgot >.<

Here it is, for reference, and sorry about the noise, https://github.com/texstudio-org/texstudio/issues/3851
